fake-snippets-api/routes/api/snippets/create.ts

```diff
@@ -28,9 +29,25 @@ export default withRouteSpec({
     circuit_json,
     dts,
   } = req.jsonBody
+
   if (!unscoped_name) {
     unscoped_name = `untitled-${snippet_type}-${ctx.db.idCounter + 1}`
   }
+
+  // Check if snippet already exists for the user
+  const existingSnippet = ctx.db.snippets.find(
+    (snippet) =>
+      snippet.unscoped_name === unscoped_name &&
+      snippet.owner_name === ctx.auth.github_username,
+  )
+
+  if (existingSnippet) {
+    return ctx.json({
+      ok: false,
+      error: "This snippet has already been forked in your account.",
```

Comment:
```suggestion
      error: "You have already forked this snippet in your account.",
```
I think this is a better error message